crypto/symmetric/symmetric.go

```diff
@@ -30,52 +40,76 @@ func New() *Key {
 	return k
 }
 
-// FromBytes returns a key by wrapping k.
+// FromBytes returns a key by decoding bytes.
 func FromBytes(k []byte) (*Key, error) {
-	if len(k) != 44 {
+	if len(k) != KeyBytes {
 		return nil, fmt.Errorf("invalid key")
 	}
 	return &Key{raw: k}, nil
 }
 
+// FromString returns a key by decoding a base32-encoded string.
+func FromString(k string) (*Key, error) {
+	_, b, err := mbase.Decode(k)
+	if err != nil {
+		return nil, err
+	}
+	return FromBytes(b)
+}
+
+// Marshal returns raw key bytes.
+func (k *Key) Marshal() ([]byte, error) {
+	return k.raw, nil
+}
+
+// Bytes returns raw key bytes.
+// This is cleaner than marshal when your not dealing with an interface type.
+func (k *Key) Bytes() []byte {
+	return k.raw
+}
+
+// String returns the base32-encoded string representation of raw key bytes.
+func (k *Key) String() string {
+	str, err := mbase.Encode(mbase.Base32, k.raw)
+	if err != nil {
+		panic("should not error with hardcoded mbase: " + err.Error())
+	}
+	return str
+}
+
 // Encrypt performs AES-256 GCM encryption on plaintext.
 func (k *Key) Encrypt(plaintext []byte) ([]byte, error) {
-	block, err := aes.NewCipher(k.raw[:32])
+	block, err := aes.NewCipher(k.raw[:KeyBytes])
 	if err != nil {
 		return nil, err
 	}
 	aesgcm, err := cipher.NewGCM(block)
 	if err != nil {
 		return nil, err
 	}
-	ciph := aesgcm.Seal(nil, k.raw[32:], plaintext, nil)
-	return ciph, nil
+	nonce := make([]byte, NonceBytes)
```

Comment:
Like we discussed a long time ago, there's no reason to keep the nonce _in the key_. Now, the nonce is created for each ciphertext, and then prepended to it.